framework/wazuh/results.py

```diff
@@ -0,0 +1,128 @@
+from wazuh.exception import WazuhException, WazuhInternalError
+
+
+class WazuhResult(dict):
+    """
+    Models a result returned by any framework function. This should be the class of object that every
+    framework function returns.
+    """
+
+    def __init__(self, dct, str_priority=None):
+        """
+        Initializes an instance
+
+        :param dct: map to take key-values from
+        :param str_priority: list of strings. If not None, conflicts when merging str values in the result
+         are solved taking the first value found in str_priority. I.e.: {'foo': 'KO'} and {'foo': 'OK'} results in
+         {'foo': 'KO'} if str_priority is set to ['KO', 'OK'] because 'KO' takes a higher priority level than 'OK'
+        """
+        super().__init__(dct)
+        self._str_priority = str_priority
+
+    def __or__(self, other):
+        """
+        | operator used to merge two WazuhResult objects. When merged with a WazuhException, the result is always a
+        WazuhException
+        :param other: WazuhResult or WazuhException
+        :return: a new WazuhResult or WazuhException
+        """
+        if isinstance(other, WazuhException):
+            return other
+        elif not isinstance(other, dict):
+            raise WazuhInternalError(1000, extra_message=f"WazuhResult cannot be merged with {type(other)} object")
+
+        result = deepcopy(self)
+
+        for key, field in other.items():
+            if key not in result:
+                result[key] = field
+            elif isinstance(field, dict):
+                result[key] = WazuhResult(result[key]) | WazuhResult(field)
+            elif isinstance(field, list):
+                self_field = result[key]
+                result[key] = [*self_field, *[elem for elem in field if elem not in self_field]]
+            elif isinstance(field, Number):
+                result[key] = result[key] + field
+            elif isinstance(field, str):  # str
+                if self._str_priority is not None:
+                    priorities = self._str_priority + result[key] + field
+                    result[key] = result[key] if priorities.index(result[key]) < priorities.index(field) else field
+                else:
+                    result[key] = "|".join([result[key], field]) if result[key] != field else field
+
+        return result
+
+    def to_dict(self):
+        """
+        Translates the result into a dict
+        :return: dict
+        """
+        return {
+            'str_priority': self._str_priority,
+            'result': deepcopy(self)
+        }
+
+    def limit(self, limit=database_limit, offset=0):
+        """
+        Should take the first `limit` results starting from `offset`
+
+        To be redefined in WazuhResult subclasses.
+
+        :param limit: integer. Default the value specified in wazuh.common.database_limit
+        :param offset: integer. Default 0.
+        :return: filtered WazuhResult
+        """
+        return deepcopy(self)
```

Comment:
are limit and offset parameters used here? I see this method is used only in the children class `WazuhQueryResult`, is it necessary to be defined in this one?

Reply:
I need this method to be implemented in the parent class because of this line:https://github.com/wazuh/wazuh/blob/fea7758af930f461d6a10db96ffdc6ac5844472f/framework/wazuh/cluster/dapi/dapi.py#L252

I need to delegate to WazuhResult subclass how to deal with limit operation